WooCommerce/Classes/ViewRelated/Dashboard/Settings/Themes/ThemeSettingViewModel.swift

```diff
@@ -12,19 +12,28 @@ final class ThemeSettingViewModel: ObservableObject {
     private let stores: StoresManager
     private let analytics: Analytics
     private(set) var carouselViewModel: ThemesCarouselViewModel
+    private let themeInstaller: ThemeInstallerProtocol
 
     init(siteID: Int64,
          stores: StoresManager = ServiceLocator.stores,
-         analytics: Analytics = ServiceLocator.analytics) {
+         analytics: Analytics = ServiceLocator.analytics,
+         themeInstaller: ThemeInstallerProtocol? = nil) {
         self.siteID = siteID
         self.stores = stores
         self.analytics = analytics
+        self.themeInstaller = themeInstaller ?? DefaultThemeInstaller()
         self.carouselViewModel = .init(mode: .themeSettings, stores: stores)
     }
 
     @MainActor
     func updateCurrentThemeName() async {
         loadingCurrentTheme = true
+
+        /// Attempt to install and activate any pending theme selection
+        /// to show correct current theme information
+        ///
+        await installPendingTheme()
```

Comment:
Why does it have to be part of the loading request? Should we call it at the end of the initializer instead?

Reply:
Good point. Moved this to `init` in 1ce3931faad0bc9e356a8923aa10757cdaf60042